08-rails/controllers.md

```diff
@@ -0,0 +1,140 @@
+- Dissect the _routes file_ and learn how to define routes
+- Explore the role of _controllers_ in a Rails application
+- Create a basic controller using the Rails generate command
+
+## Setting Up a Controller
+
+A Rails Controller is the central manager of a Rails application. It takes requests from the web server, pulls information from data models and makes decisions. It then hands information to the views for rendering the data back to the user.
+
+In Rails, we will build a separate class for each controller. The class is named after the type of resource it controls, and always ends with the word `Controller`. For example, in our books application, we might have a `BooksController` class and an `AuthorsController` class.
+
+The things that a controller can do are called _actions_, and each action will be an instance method on a controller class. For example, our `BooksController` might have an `index` action that sends back a list of books to the browser, and a `show` action that sends back details on one book. Each of those would be an instance method in the `BooksController`.
+
+Many of these actions are so common that we use the same names for them every time - `index` and `show` are examples of this.
+
+We'll start with just one controller (the `BooksController`) and one action (`index`). Rails can create a Controller for us with the following command:
+
+```bash
+rails generate controller Books
+```
+
+Note that `Books` (the controller name) is capitalized and pluralized.
+
+### The Controller Class
+
+Rails places controller classes in the `app/controllers` folder.  In the case of the Controller we just generated it will be in the `books_controller.rb` class.
+
+**Question:** Does this follow our standard Ruby naming convention for classes and files?
+
+Opening the Controller file you will find the code below:
+
+```ruby
+# app/controllers/books_controller.rb
+class BooksController < ApplicationController
+end
+```
+
+Note that while we typed `rails generate controller Books`, Rails called the controller `BooksController`. This kind of shorthand is common in Rails - you'll quickly get used to it.
+
+**Question:** What does `< ApplicationController` mean? How might this impact our class?
+
+We've got a controller but no actions yet. To add the `index` action, create an empty method called `index` inside the controller class:
+
+```ruby
+# app/controllers/books_controller.rb
+class BooksController < ApplicationController
+  def index
+  end
+end
+```
+
+This method won't stay empty for long, but first we've got a little more work to do.
+
+## Adding a Route
+
+We've now created a controller (the `BooksController` class) and given it an action (the `index` method). However, Rails doesn't automatically know how to use this controller; we have to tell it.
+
+To do so, we will introduce a new piece of Rails: the router. The router does its job with the help of a special configuration file that lives at
```

Comment:
Since we're introducing the router in this lesson as well, is there a way we can highlight it as a major thing? I don't necessarily want to conflate Controllers and Routers though, but this just feels like A Big Thing To Get

Reply:
An extreme would be renaming the lesson, another option may be changing headers

Reply:
This lesson was originally called `routes-and-controllers.md`. My goal was to remove as much of the router info as I can until that's really useful, and just focus on controllers to start.

However, I see your point - letting them know that this routefile is important and we'll be coming back to it is valuable.